ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -328,7 +328,6 @@ getIdeOptionsIO ide = do
 -- for the version of that value.
 lastValueIO :: IdeRule k v => ShakeExtras -> k -> NormalizedFilePath -> IO (Maybe (v, PositionMapping))
```

Comment:
Interestingly, I think this entire function is close to being in STM. Maybe just `persistentKeys` blocking it. Not sure if that's desirable.

Reply:
I think this one is firmly in `IO`